lib/utils/ec2.go

```diff
@@ -84,3 +88,83 @@ func IsEC2NodeID(id string) bool {
 func NodeIDFromIID(iid *imds.InstanceIdentityDocument) string {
 	return iid.AccountID + "-" + iid.InstanceID
 }
+
+// InstanceMetadata is an interface for fetching information from EC2 instance
+// metadata.
+type InstanceMetadata interface {
```

Comment:
I know we don't always follow this, but I like the idea that [interfaces generally belong in the package that uses values of the interface type, not the package that implements those values](https://github.com/golang/go/wiki/CodeReviewComments#interfaces). You can just export the concrete type, if the caller wants to mock it they can define the interface in their own package